perception/autoware_shape_estimation/lib/tensorrt_shape_estimator.cpp

```diff
@@ -0,0 +1,338 @@
+                       {11.194943, 2.9501154, 3.4918275},     {12.275775, 2.9231303, 3.87086},
+                       {0.80057803, 0.5983815, 1.27450867},   {1.76282397, 0.59706367, 1.73698127},
+                       {16.17150617, 2.53246914, 3.53079012}, {3.64300781, 1.54298177, 1.92320313}};
+}
+
+bool TrtShapeEstimator::inference(
+  const DetectedObjectsWithFeature & input, DetectedObjectsWithFeature & output)
+{
+  if (!trt_common_->isInitialized()) {
+    return false;
+  }
+
+  bool result = false;
+
+  for (size_t i = 0; i < input.feature_objects.size(); i += batch_size_) {
+    DetectedObjectsWithFeature input_batch;
+    input_batch.header = input.header;
+
+    for (size_t j = 0; j < batch_size_ && (i + j) < input.feature_objects.size(); ++j) {
+      input_batch.feature_objects.push_back(input.feature_objects[i + j]);
+    }
+
+    preprocess(input_batch);
+    DetectedObjectsWithFeature output_batch;
+    result = feed_forward_and_decode(input_batch, output_batch);
+
+    output.feature_objects.insert(
+      output.feature_objects.end(), output_batch.feature_objects.begin(),
+      output_batch.feature_objects.end());
+  }
+
+  return result;
+}
+
+void TrtShapeEstimator::preprocess(const DetectedObjectsWithFeature & input)
+{
+  auto input_dims_pc = trt_common_->getBindingDimensions(0);
+  int batch_size = static_cast<int>(input.feature_objects.size());
+
+  const auto input_chan = static_cast<float>(input_dims_pc.d[1]);
+  const auto input_pc_size = static_cast<float>(input_dims_pc.d[2]);
+
+  auto input_dims_one_hot = trt_common_->getBindingDimensions(1);
+  const auto input_one_hot_size = static_cast<float>(input_dims_one_hot.d[1]);
+
+  int volume_pc = batch_size * input_chan * input_pc_size;
+  input_pc_h_.resize(volume_pc);
+
+  int volume_one_hot = batch_size * input_one_hot_size;
+  input_one_hot_h_.resize(volume_one_hot);
+
+  // fill point cloud
+  for (size_t i = 0; i < input.feature_objects.size(); i++) {
+    const auto & feature_object = input.feature_objects[i];
+
+    int point_size_of_cloud =
+      feature_object.feature.cluster.width * feature_object.feature.cluster.height;
+
+    if (point_size_of_cloud <= input_pc_size) {
+      int offset = 0;
+      for (sensor_msgs::PointCloud2ConstIterator<float> x_iter(feature_object.feature.cluster, "x"),
+           y_iter(feature_object.feature.cluster, "y"), z_iter(feature_object.feature.cluster, "z");
+           x_iter != x_iter.end(); ++x_iter, ++y_iter, ++z_iter) {
+        input_pc_h_[i * input_chan * input_pc_size + 0 + offset] = -1 * (*y_iter);
+        input_pc_h_[i * input_chan * input_pc_size + 512 + offset] = -1 * (*z_iter);
+        input_pc_h_[i * input_chan * input_pc_size + 1024 + offset] = *x_iter;
+        offset++;
+      }
+
+      int iter_count = static_cast<int>(input_pc_size) / point_size_of_cloud;
+      int remainer_count = static_cast<int>(input_pc_size) % point_size_of_cloud;
```

Comment:
```suggestion
      int remainder_count = static_cast<int>(input_pc_size) % point_size_of_cloud;
```

Spell check failed, so can you change `remainer` to another word? I think `reminder` is better.
https://github.com/autowarefoundation/autoware.universe/actions/runs/10196767728/job/28208175986?pr=7860